packages/core/strapi/lib/core/app-configuration/load-config-file.js

```diff
@@ -30,6 +30,10 @@ const loadJSONFile = file => {
 const loadFile = file => {
   const ext = path.extname(file);
 
+  if(ext !== '.js' && ext !== '.json' && ext !== '.ts') {
```

Comment:
What we want to do is actually not load the unintended files. to do that you can do the following change in 

```diff
 const fs = require('fs');
 const loadFile = require('./load-config-file');

+const VALID_EXTENSIONS = ['.js', '.json']
+
 module.exports = dir => {
   if (!fs.existsSync(dir)) return {};

   return fs
     .readdirSync(dir, { withFileTypes: true })
-    .filter(file => file.isFile())
+    .filter(file => file.isFile() && VALID_EXTENSIONS.includes(path.extname(file.name)))
     .reduce((acc, file) => {
       const key = path.basename(file.name, path.extname(file.name));

```

Reply:
Hi @alexandrebodin ,
only .js and .json should be accepted or also .ts extension?

Reply:
Strapi only loads js or json files :) ts files will be pre compiled to js so no need to keep it